grpc/src/main/java/io/stargate/grpc/service/Service.java

```diff
@@ -143,6 +156,8 @@ public void executeQuery(Query query, StreamObserver<Response> responseObserver)
       Connection connection = newConnection(authenticationSubject.asUser());
       QueryParameters queryParameters = query.getParameters();
 
+      CompletableFuture<Prepared> tracingPrepareQuery = prepareTracingQuery(connection);
```

Comment:
we could prepare this query only if tracing is enabled. However, there is only one prepared query so overhead is low, and paid only once (during the first query). I am not sure if introducing the if is worth complicating the code.

Reply:
Do tracing queries need to be prepared at all? Maybe just use `DataStore` and it's handled automatically.